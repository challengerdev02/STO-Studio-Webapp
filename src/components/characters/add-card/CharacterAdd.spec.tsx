import { cleanup, render } from '@testing-library/react';
import { CharacterAdd } from './index';

describe('CharacterAdd', function () {
  afterEach(cleanup);

  const setup = (
    extraProps: Record<string, any> = { handleCreateScene: () => {} }
  ) => {
    const utils = render(
      <CharacterAdd onCreateCharacter={() => {}} {...extraProps} />
    );
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should test that the card is rendered', function () {
    const { container } = setup();
    expect(container).toBeDefined();
  });

  it('should test the test "Add new scene" to be in the document', function () {
    const { getByTestId } = setup();
    expect(getByTestId('add')).toBeInTheDocument();
    expect(getByTestId('add').textContent).toBe('Add new character');
  });
});
