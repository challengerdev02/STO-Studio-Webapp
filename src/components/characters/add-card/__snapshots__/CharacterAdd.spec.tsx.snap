// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterAdd should match with snapshot 1`] = `
<div>
  <div
    style="cursor: pointer;"
  >
    <div
      class="ant-card sc-gsDKAQ jFIuNG"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="sc-hKwDye dEuGEj"
        >
          <i
            class="sc-dkPtRN dKVEsJ  mc-plus-2-line"
          />
        </div>
        <div
          class="sc-eCImPb apvtg"
          data-testid="add"
        >
          Add new character
        </div>
        <div
          class="sc-jRQBWg jcdHyb"
        >
          (
          <span
            class="sc-gKclnd isKiau"
          >
            Character:
          </span>
           jpg, png, svg, or gif)
        </div>
      </div>
    </div>
  </div>
</div>
`;
